MAF requires a workaround, due to a BioPerl issue for develop branch

greatfireball opened the pull request:
Due to the BioPerl issue (bioperl/bioperl-live#147) I need to apply a workaround for MAF alignment imports.

Additionally, only required alignments are calculated by lastz

closes #52

greatfireball committed:
Introducing a flag to fix MAF revcomp issue while link import

This flag is used while import of all links to determine if MAF was
the input file. In that case, we need to adjust the location of hits
on (-)strand. This seems to be a bug in BioPerl and might be fixed in
future.

An issue for BioPerl was already opened:

	https://github.com/bioperl/bioperl-live/issues/147

greatfireball committed:
Added a test for AliTV::Genome::_store_feature()

greatfireball committed:
Implemented a new AliTV::Genome::_store_feature()

greatfireball committed:
Added a second link to test set of _store_feature

greatfireball committed:
Added an identical link to test set of _store_feature

greatfireball committed:
Added two identical links to cause exception in _store_feature

greatfireball committed:
Implemented the exception if two identical links are existing

greatfireball committed:
Workaround if Bio::AlignIO::format() method not present

greatfireball committed:
Added a test for AliTV::Genome::seq_exists() method

greatfireball committed:
Added a basic AliTV::Genome::seq_exists() method

greatfireball committed:
Added a test set for AliTV::Genome::seq_exists() method

greatfireball committed:
Implemented AliTV::Genome::seq_exists() method for unique seq names

greatfireball committed:
Added a test set for AliTV::Genome::seq_exists() method with generated uniq names

greatfireball committed:
Implemented AliTV::Genome::seq_exists() method for all seq names

greatfireball committed:
Added a test for AliTV::Genome::fix_maf_revcomp() method

greatfireball committed:
Added a basic AliTV::Genome::fix_maf_revcomp() method

greatfireball committed:
Added a test for forward strand maf issues

greatfireball committed:
Implemented fix_maf_revcomp for forward strand

greatfireball committed:
Added a test for reverse strand maf issues

greatfireball committed:
Implemented fix_maf_revcomp for reverse strand

greatfireball committed:
Link features are now corrected for MAF revcomp issue

greatfireball committed:
Feature import of -1 features guarantees start > end and vice versa

greatfireball committed:
Avoid multiple links at the same position

greatfireball committed:
Corrected calculation of other strands start and end coordinates

greatfireball committed:
Added corrected expected values to maf-workaround test

greatfireball committed:
Merge pull request #70 from AliTVTeam/hotfix/renaming_sequence_names

Sequence names are corrected for alignments

greatfireball committed:
Updated the expected alignment results

greatfireball committed:
Checking the number of expected alignments

greatfireball committed:
Retitled the deep comparison test for alignments

greatfireball committed:
Lastz avoiding unrequired alignments

greatfireball committed:
Lastz: Added some status messages

greatfireball committed:
Bumped patch level to v0.1.7

review-ninja commented:
<a href="https://app.review.ninja/AliTVTeam/AliTV-perl-interface/pull/73" target="_blank"><img src="https://app.review.ninja/41874017/pull/73/badge" alt="ReviewNinja"/>

greatfireball committed:
Fixed calculation of number of required alignments

greatfireball committed:
Removed newlines from log messages

greatfireball committed:
Added some documentation to the AliTV::Genome class

greatfireball committed:
Added tests for different internal seq_id methods

greatfireball committed:
Added methods seq_id methods to Genome class

Added the methods
   Genome::_get_seq_ids()
   Genome::_get_uniq_seq_ids()
   Genome::_get_orig_seq_ids()

to genome class.

greatfireball committed:
Added test for valid Genome::_get_uniq_seq_ids() result

greatfireball committed:
Implemented the Genome::_get_uniq_seq_ids

greatfireball committed:
Added test for valid Genome::_get_orig_seq_ids() result

greatfireball committed:
Implemented the Genome::_get_orig_seq_ids()

greatfireball committed:
Added test for valid Genome::_get_seq_ids() behavior

greatfireball committed:
Implemented the Genome::_get_seq_ids() method

greatfireball committed:
Updated Genome class to use _get_seq_ids methods

greatfireball committed:
New implementation of Genome::set_uniq_seq_names()

greatfireball committed:
Added test for mapping of original to unique sequence names and reverse

greatfireball committed:
Added methods for mapping of original to unique sequence names and reverse

greatfireball committed:
Added test for mapping of original to unique sequence IDs

greatfireball committed:
Implemented method AliTV::Genome::_orig_id_to_uniq_id()

greatfireball committed:
Added test for mapping of unique sequence IDs back to original sequence IDs

greatfireball committed:
Implemented method AliTV::Genome::_uniq_id_to_orig_id()

greatfireball committed:
Updated AliTV::Genome::get_chromosomes to use mapping methods
